torch/distributed/rpc/api.py

```diff
@@ -103,18 +161,19 @@ def shutdown(graceful=True):
         >>> # wait for worker 0 to finish work, and then shutdown.
         >>> rpc.shutdown()
     """
-    global _agent
-    if _agent:
-        if graceful:
-            _wait_all_workers()
-        _destroy_rref_context(_ignore_rref_leak)
-        _agent.shutdown()
-        # clean up python rpc handler in shutdown(), see comments in
-        # PythonRpcHandler::cleanup(), call it in python API because the
-        # cleanup() function has python dependency, it assumes python
-        # interpreter exists
-        _cleanup_python_rpc_handler()
-        _agent = None
```

Comment:
why not setting _agent to None to allow calling shutdown multiple times? If we disallow calling it twice, let's add that to the doc. Besides, the error message also needs to change. I believe in the current current, users would see `Can not call wait_all_workers() twice.` if they call `shutdown` twice. But since there is no public API `wait_all_workers`, it might be confusing.

Reply:
Accidentally removed this. This is the reason of test failures.